src/amo/components/PermissionsCard/permissions.js

```diff
@@ -0,0 +1,231 @@
+      clipboardRead: i18n.gettext('Get data from the clipboard'),
+      clipboardWrite: i18n.gettext('Input data to the clipboard'),
+      devtools: i18n.gettext('Extend developer tools to access your data in open tabs'),
+      downloads: i18n.gettext('Download files and read and modify the browser’s download history'),
+      'downloads.open': i18n.gettext('Open files downloaded to your computer'),
+      find: i18n.gettext('Read the text of all open tabs'),
+      geolocation: i18n.gettext('Access your location'),
+      history: i18n.gettext('Access browsing history'),
+      management: i18n.gettext('Monitor extension usage and manage themes'),
+      // In Firefox the following message replaces the name "Firefox" with the
+      // current brand name, e.g., "Nightly", but we do not need to do that.
+      nativeMessaging: i18n.gettext('Exchange messages with programs other than Firefox'),
+      notifications: i18n.gettext('Display notifications to you'),
+      pkcs11: i18n.gettext('Provide cryptographic authentication services'),
+      proxy: i18n.gettext('Control browser proxy setting'),
+      privacy: i18n.gettext('Read and modify privacy settings'),
+      sessions: i18n.gettext('Access recently closed tabs'),
+      tabs: i18n.gettext('Access browser tabs'),
+      tabHide: i18n.gettext('Hide and show browser tabs'),
+      topSites: i18n.gettext('Access browsing history'),
+      unlimitedStorage: i18n.gettext('Store unlimited amount of client-side data'),
+      webNavigation: i18n.gettext('Access browser activity during navigation'),
+      allUrls: i18n.gettext('Access your data for all websites'),
+      wildcard: (domain) => {
+        return i18n.sprintf(
+          i18n.gettext('Access your data for sites in the %(domain)s domain'),
+          { domain }
+        );
+      },
+      // Note that we do not need to use ngettext for this as count will never
+      // be less than 2.
+      tooManyWildcards: (count) => {
+        return i18n.sprintf(
+          i18n.gettext('Access your data in %(count)s other domains'),
+          { count: i18n.formatNumber(count) },
+        );
+      },
+      oneSite: (site) => {
+        return i18n.sprintf(
+          i18n.gettext('Access your data for %(site)s'),
+          { site }
+        );
+      },
+      // Note that we do not need to use ngettext for this as count will never
+      // be less than 2.
+      tooManySites: (count) => {
+        return i18n.sprintf(
+          i18n.gettext('Access your data on %(count)s other sites'),
+          { count: i18n.formatNumber(count) },
+        );
+      },
+    };
+  }
+
+  // Get a list of permissions from the correct platform file.
+  getCurrentPermissions(
+    addon: AddonType,
+    userAgentInfo: UserAgentInfoType
+  ): Array<string> {
+    const agentOsName =
+      userAgentInfo.os.name && userAgentInfo.os.name.toLowerCase();
+    const platform = userAgentOSToPlatform[agentOsName];
```

Comment:
I added a warning, but for some reason it's made Flow upset. I'm seeing this:

```
change:src/amo/components/PermissionsCard/permissions.js
Error: src/amo/components/PermissionsCard/permissions.js:90
 90:       log.debug(`No platform exists for user agent OS "${agentOsName}"`);
                                                              ^^^^^^^^^^^ undefined. This type cannot be coerced to
 90:       log.debug(`No platform exists for user agent OS "${agentOsName}"`);
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ string
```

which I find confusing for two reasons:

1. It's actually entirely valid to pass `undefined` into a template string, although it will result in the string `undefined` instead of an empty string, which I think is fine in this case.
2. I was surprised that Flow is complaining about this now, when I already had another log statement which uses `agentOsName` in a template string and it didn't complain about that. I see now that the older statement uses `oneLineTrim`, and if I wrap the new log statement in that then Flow ceases complaining, but I don't need `oneLineTrim` for this one, and I don't really understand why that makes a difference.

Reply:
Relying on the string `undefined` is relying on JS quirks which is why Flow forbids it. You could do something like `${agentOsName || '<undefined agentOsName>'}` which would also provide a debugging hint.

Reply:
Well, I wasn't going to rely on the string being `undefined`, it's just that the message would be fine with either `underfined` or an empty string, but I guess Flow doesn't know that.

Reply:
Ironically, adding that code makes the message so long that I now _do_ need to use `oneLineTrim`. 😉